test/WebJobs.Script.Tests/Diagnostics/KubernetesEventGeneratorTests.cs

```diff
@@ -24,7 +26,7 @@ public KubernetesEventGeneratorTests()
                 _events.Add(s);
             };
 
-            _generator = new KubernetesEventGenerator(writer);
+            _generator = new KubernetesEventGenerator(Options.Create(new ConsoleLoggingOptions()), writer);
```

Comment:
If you follow my suggestion above, you can add a test that verifies no logs are written when logs are disabled.